controllers/confidential_handler.go

```diff
@@ -0,0 +1,53 @@
+package controllers
+
+import (
+	"context"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func patchConfigMap(client client.Client, name string, namespace string, data map[string]string) error {
+	configMap := &corev1.ConfigMap{}
+	if err := client.Get(context.TODO(), types.NamespacedName{
+		Name:      name,
+		Namespace: namespace,
+	}, configMap); err != nil {
+		return err
+	}
+
+	for key, value := range data {
+		configMap.Data[key] = value
+	}
+
+	return client.Update(context.TODO(), configMap)
+}
+
+func toggleFeatureConfidential(client client.Client, enabled bool) error {
+
+	var podvmData map[string]string
+	var peerpodsData map[string]string
+
+	if enabled {
```

Comment:
Few questions:
- Should the code check if `enablePeerPods == true` before (re)setting the peer-pods cm ?
- Is there a need to check and reset the `AZURE_IMAGE_ID`, since if this is set then image won't be generated.
- Should the code be structured in a way to handle other providers when needed instead of hard-coding it to azure specifics?

Reply:
> Should the code be structured in a way to handle other providers when needed instead of hard-coding it to azure specifics?

Wouldn't it make sense to implement this logic be in the image creation scripts?

And some more questions:
Will we support switching confidential/non-confidential  at runtime?
what about verifying the instance type user sets is confidential? should we enforce it? (maybe in the scripts?) what happens if it's non confidential ? (IIRC DISABLECVM=true) will cause it to fail